### DISCUSSION TITLE
Fixed height slider with varying width images

### DISCUSSION THREAD

**Discussion by johnlewisdt** [nolimits4web/swiper#4490] [Q&A]
Hi, I'm struggling to achieve the above and its cost me a lot of time.

I have a range of pictures, some square, some portrait, some landscape. **All are 200px high**. 

Trouble is, Swiper wants to automatically spread them over x width, pull them out of proportion and stretch them beyond recognition to fit equal rectangles. 

I just want to see all of the image, whether it is 800px wide or 10px wide, with an equal 1em gutter.

What do I do to get around this? Is there a magic combo of params? I don't seem to be having much luck randomly combining parameters, they all seem to clash with each other and waste me hours trying to get it to behave. 

Any help much appreciated,

J

**Comment by vltansky** [comment]
A live example would be really helpful, create [codesandbox](https://codesandbox.io/) or fork one from https://swiperjs.com/demos#thumbs-gallery.
It seems to me as it can be solved with CSS. Maybe set image as background and use `background-size: contain` or if its `<img>` you can use `object-fit: contain`

**Comment by linonetwo** [comment]
Yes yes, this is very hard to achieve, I tried, but not succeed https://github.com/nolimits4web/swiper/discussions/4646

Seems width is calculated by JS, not CSS, so this may need a plugin.
